### PR TITLE
Fixed path filtering by maxLatency

### PR DESCRIPTION
**Pull request by Nexgen90**
Added check for spur paths weight and only if it is compatible - add it to the potentialKthShortestPaths list.
Extended Path-entity with isBackupPath parameter.

Related to #4525

### CODE REVIEW

File: src-java/kilda-pce/src/main/java/org/openkilda/pce/finder/BestWeightAndShortestPathFinder.java [hunk 1]
```diff
@@ -155,7 +155,8 @@ public List<List<Edge>> findNPathsBetweenSwitches(
             WeightFunction weightFunction, long maxWeight, long backUpMaxWeight) throws UnroutableFlowException {
         Node end = network.getSwitch(endSwitchId);
         return findNPathsBetweenSwitches(network, startSwitchId, endSwitchId, count, weightFunction,
-                (Node start) -> findOneDirectionPath(start, end, weightFunction, maxWeight, backUpMaxWeight));
+                (Node start) -> findOneDirectionPath(start, end, weightFunction, maxWeight, backUpMaxWeight),
+                maxWeight);
```
Comment (dpoltavets) [inline src-java/kilda-pce/src/main/java/org/openkilda/pce/finder/BestWeightAndShortestPathFinder.java:159]: What about `backUpMaxWeight`?

**Review by dpoltavets — CHANGES_REQUESTED** [review]

File: src-java/kilda-pce/src/main/java/org/openkilda/pce/finder/BestWeightAndShortestPathFinder.java (hunk 1 above)
Comment (Nexgen90) [reply src-java/kilda-pce/src/main/java/org/openkilda/pce/finder/BestWeightAndShortestPathFinder.java]: Fixed

File: src-java/base-topology/base-messaging/src/main/java/org/openkilda/messaging/info/network/Path.java [hunk 2]
```diff
@@ -37,11 +37,16 @@ public class Path implements Serializable {
     @JsonProperty("nodes")
     private List<PathNodePayload> nodes;
 
+    @JsonProperty("isBackupPath")
```
Comment (dpoltavets) [inline src-java/base-topology/base-messaging/src/main/java/org/openkilda/messaging/info/network/Path.java:40]: Please use snake_case for json property naming.

File: src-java/base-topology/base-messaging/src/main/java/org/openkilda/messaging/payload/network/PathDto.java [hunk 3]
```diff
@@ -43,20 +43,25 @@ public class PathDto {
     @JsonProperty("nodes")
     private List<PathNodePayload> nodes;
 
-    public PathDto(Long bandwidth, Duration latency, List<PathNodePayload> nodes) {
-        this(bandwidth, latency.toNanos(), latency.toNanos(), latency.toMillis(), nodes);
+    @JsonProperty("isBackupPath")
```
Comment (dpoltavets) [inline src-java/base-topology/base-messaging/src/main/java/org/openkilda/messaging/payload/network/PathDto.java:46]: The same here

File: src-java/kilda-pce/src/main/java/org/openkilda/pce/finder/BestWeightAndShortestPathFinder.java [hunk 4]
```diff
@@ -233,34 +245,39 @@ private List<List<Edge>> findNPathsBetweenSwitches(
...
         return bestPaths;
     }
 
-    private List<Edge> getBestPotentialKthShortestPath(Set<List<Edge>> potentialKthShortestPaths,
-                                                       List<List<Edge>> bestPaths,
-                                                       WeightFunction weightFunction) {
-        List<Edge> bestKthShortestPath = new ArrayList<>();
-
+    private FindOneDirectionPathResult getBestPotentialKthShortestPath(
+            Set<FindOneDirectionPathResult> potentialKthShortestPaths,
```
Comment (dpoltavets) [inline src-java/kilda-pce/src/main/java/org/openkilda/pce/finder/BestWeightAndShortestPathFinder.java:259]: Please format the method signature in the same style as before.

File: src-java/kilda-pce/src/test/java/org/openkilda/pce/impl/MaxLatencyPathComputationStrategyBaseTest.java [hunk 5]
```diff
@@ -273,6 +280,26 @@ public void maxLatencyShouldChooseCorrectWayTest() throws Exception {
         assertEquals(path.getForward().getSegments().get(1).getSrcSwitchId(), new SwitchId("00:03"));
     }
 
+    @Test
+    public void shouldNotFindPathsGreaterThenMaxLatency() throws Exception {
```
Comment (dpoltavets) [inline src-java/kilda-pce/src/test/java/org/openkilda/pce/impl/MaxLatencyPathComputationStrategyBaseTest.java:284]: Please also add a test for the maxLatencyTier2 case.

**Review by dpoltavets — CHANGES_REQUESTED** [review]

File: src-java/base-topology/base-messaging/src/main/java/org/openkilda/messaging/info/network/Path.java (hunk 2 above)
Comment (Nexgen90) [reply src-java/base-topology/base-messaging/src/main/java/org/openkilda/messaging/info/network/Path.java]: Fixed

File: src-java/base-topology/base-messaging/src/main/java/org/openkilda/messaging/payload/network/PathDto.java (hunk 3 above)
Comment (Nexgen90) [reply src-java/base-topology/base-messaging/src/main/java/org/openkilda/messaging/payload/network/PathDto.java]: Fixed

File: src-java/kilda-pce/src/main/java/org/openkilda/pce/finder/BestWeightAndShortestPathFinder.java (hunk 4 above)
Comment (Nexgen90) [reply src-java/kilda-pce/src/main/java/org/openkilda/pce/finder/BestWeightAndShortestPathFinder.java]: Fixed

File: src-java/kilda-pce/src/test/java/org/openkilda/pce/impl/MaxLatencyPathComputationStrategyBaseTest.java (hunk 5 above)
Comment (Nexgen90) [reply src-java/kilda-pce/src/test/java/org/openkilda/pce/impl/MaxLatencyPathComputationStrategyBaseTest.java:284]: Done

**Review by dpoltavets — APPROVED** [review]

**Review by niksv — APPROVED** [review]

**Pull request merged**